code/modules/admin/banjob.dm

```diff
@@ -1,159 +1,62 @@
+		return JB.reason
 
 	return null
 
-/proc/jobban_loadbans()
-	if(!SSdbcore.IsConnected())
-		log_world("Database connection failed. Job bans not loaded.")
-		return
-
-	//Job permabans
-	var/datum/db_query/permabans = SSdbcore.NewQuery("SELECT ckey, job FROM ban WHERE bantype = 'JOB_PERMABAN' AND isnull(unbanned)")
-
-	if(!permabans.warn_execute(async=FALSE))
-		qdel(permabans)
-		return FALSE
-
-	while(permabans.NextRow())
-		var/ckey = permabans.item[1]
-		var/job = permabans.item[2]
-		GLOB.jobban_keylist.Add("[ckey] - [job]")
-		jobban_assoc_insert(ckey, job)
-
-	qdel(permabans)
-
-	// Job tempbans
-	var/datum/db_query/tempbans = SSdbcore.NewQuery("SELECT ckey, job FROM ban WHERE bantype = 'JOB_TEMPBAN' AND isnull(unbanned) AND expiration_time > Now()")
+// Gets all the job bans for a ckey incase they are offline
+/proc/get_jobbans_for_offline_ckey(ckey)
+	var/datum/db_query/query = SSdbcore.NewQuery({"
+		SELECT job FROM ban
+		WHERE ckey LIKE :ckey AND ((bantype LIKE 'JOB_TEMPBAN' AND expiration_time > NOW()) OR (bantype LIKE 'JOB_PERMABAN')) AND ISNULL(unbanned)
+		ORDER BY bantime DESC LIMIT 100"},
+		list("ckey" = ckey)
+	)
 
-	if(!tempbans.warn_execute(async=FALSE))
-		qdel(tempbans)
+	if(!query.warn_execute())
+		qdel(query)
 		return FALSE
 
-	while(tempbans.NextRow())
-		var/ckey = tempbans.item[1]
-		var/job = tempbans.item[2]
-		GLOB.jobban_keylist.Add("[ckey] - [job]")
-		jobban_assoc_insert(ckey, job)
-
-	qdel(tempbans)
-
-/proc/jobban_savebanfile()
-	var/savefile/S=new("data/job_full.ban")
-	S["keys[0]"] << GLOB.jobban_keylist
-
-/proc/jobban_unban(mob/M, rank)
-	jobban_remove("[M.ckey] - [rank]")
+	var/list/jobs = list()
 
-/proc/jobban_unban_client(ckey, rank)
-	jobban_remove("[ckey] - [rank]")
+	while(query.NextRow())
+		jobs += query.item[1]
 
-/proc/jobban_remove(X)
-	for(var/i = 1; i <= length(GLOB.jobban_keylist); i++)
-		if( findtext(GLOB.jobban_keylist[i], "[X]") )
-			// This need to be here, instead of jobban_unban, due to direct calls to jobban_remove
-			if(GLOB.jobban_regex.Find(X))
-				var/ckey = GLOB.jobban_regex.group[1]
-				var/rank = GLOB.jobban_regex.group[2]
-				if(GLOB.jobban_assoclist[ckey] && GLOB.jobban_assoclist[ckey][rank])
-					GLOB.jobban_assoclist[ckey] -= rank
-				else
-					log_runtime(EXCEPTION("Attempted to remove non-existent job ban: [X]"))
-			else
-				log_runtime(EXCEPTION("Failed to remove malformed job ban from associative list: [X]"))
-			GLOB.jobban_keylist.Remove(GLOB.jobban_keylist[i])
-			return 1
-	return 0
+	return jobs
 
-/mob/verb/displayjobbans()
+/client/verb/displayjobbans()
 	set category = "OOC"
 	set name = "Display Current Jobbans"
 	set desc = "Displays all of your current jobbans."
 
-	if(!client || !ckey)
-		return
+	// Ok. I know this verb here is scoped to client, I know.
+	// But sometimes when executing, the src will be a mob
+	// I have no idea why, but this is a workaround.
+	jbh.reload_jobbans(usr.client)
 
 	var/is_actually_banned = FALSE
```

Comment:
Can get rid of this var since it seems to only be used in the one `if`

Reply:
Will do